src/test/kotlin/com/wolt/arrow/detekt/rules/NoRaiseBindableValueAsStatementTest.kt

```diff
@@ -0,0 +1,414 @@
+package com.wolt.arrow.detekt.rules
+
+import io.gitlab.arturbosch.detekt.api.Config
+import io.gitlab.arturbosch.detekt.rules.KotlinCoreEnvironmentTest
+import io.gitlab.arturbosch.detekt.test.lintWithContext
+import io.kotest.matchers.collections.shouldHaveSize
+import org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+
+@KotlinCoreEnvironmentTest
+internal class NoRaiseBindableValueAsStatementTest(private val env: KotlinCoreEnvironment) {
+
+    @Nested
+    @DisplayName("bindable variations")
+    inner class BindableVariations {
+        @Test
+        fun `reports unbound Either`() {
+            val code = """
+                import arrow.core.Either
+                import arrow.core.raise.Effect
+                import arrow.core.raise.effect
+                
+                fun test(): Effect<Throwable, Int> = effect {
+                    Either.Right(1)
+                    1
+                }
+            """
+            val findings = NoEffectScopeBindableValueAsStatement(Config.empty).lintWithContext(env, code)
+            findings shouldHaveSize 1
+        }
+
+        @Test
+        fun `reports unbound Validated`() {
+            val code = """
+                import arrow.core.Validated
+                import arrow.core.raise.Effect
+                import arrow.core.raise.effect
+                
+                fun test(): Effect<Throwable, Int> = effect {
+                    Validated.Valid(1)
+                    1
+                }
+            """
+            val findings = NoEffectScopeBindableValueAsStatement(Config.empty).lintWithContext(env, code)
+            findings shouldHaveSize 1
+        }
```

Comment:
Wasn't `Validated` removed in Arrow 2? At least with `2.0.0-SNAPSHOT` this test case fails due to not having `arrow.core.Validated` 🤔